pkg/skaffold/k8sjob/util.go

```diff
@@ -183,3 +184,22 @@ func isRetryableErr(k8sErr error) bool {
 	}
 	return isRetryable
 }
+
+func CheckIfPullImgErr(pod *corev1.Pod, jobName string) error {
+	for _, cs := range pod.Status.ContainerStatuses {
+		if cs.State.Waiting == nil {
+			continue
+		}
+		if checkIsPullImgErr(cs.State.Waiting.Reason) {
+			return fmt.Errorf("creating pod for job %v: %v", jobName, cs.State.Waiting.Reason)
```

Comment:
should this be creating `container` instead of `pod` ?

Reply:
Yeap, changed.